bytecomp/matching.ml

```diff
@@ -3002,15 +3492,19 @@ let simple_for_let loc param pat body =
 let rec map_return f = function
```

Comment:
I think it would make sense to have `f` expect a `Lambda.block` argument, and then pass the nearest block's location to `map_return`, just like you did for `Cmmgen.unbox_float`.

Reply:
This is probably right, so I've changed it---see caveat in the next comment though.